### DISCUSSION TITLE
page.get_images() returns the images from other pages

### DISCUSSION THREAD

**Discussion by lokman2k5** [pymupdf/PyMuPDF#3898] [Looking for help] [answered]
with a certain PDF file, [vitas.pdf](https://github.com/user-attachments/files/17152906/vitas.pdf), when i use page.get_images() on the first page for example, it returns every image in the PDF. I have tested with other PDFs and it works correctly and only returns images from the same page. How can I solve this?
Here is my code: 
`import fitz
doc = fitz.open("vitas.pdf")
page = doc[0]
for image in page.get_images():
    print(image[0])
    rawImg = doc.extract_image(image[0])
    imageFile = f"image{image[0]}.{rawImg['ext']}"
    imgOut = open(imageFile, 'wb')
    imgOut.write(rawImg['image'])
    print(f"{imageFile} is saved")
`

**Comment by JorjMcKie** [comment, marked as answer]
This method returns the contents of the respective part of the page's object definition. It is not a statement about what the page in fact shows.
The PDF creator can enter whatever in that array.
You can use `page.get_image_info()` instead.
Or use `page.clean_contents()` before execution. This will make a run through the page's source code and synchronize its object definitions with what really happens during page dislay.

**Reply by lokman2k5** [reply]
thank you. clean_contents() did the job!
